src/Microsoft.ML.Core/Environment/HostEnvironmentBase.cs

```diff
@@ -567,4 +571,64 @@ public virtual void PrintMessageNormalized(TextWriter writer, string message, bo
         else if (!removeLastNewLine)
             writer.WriteLine();
     }
+
+    /// <summary>
+    /// Adds a new runtime option. Throws if option already exists.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="name"></param>
+    /// <param name="value"></param>
+    public void AddOption<T>(string name, T value)
+    {
+        Options.Add(name, value);
+    }
+
+    /// <summary>
+    /// Trys to add a new runtime option.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="name"></param>
+    /// <param name="value"></param>
+    /// <returns>True if successful. False otherwise.</returns>
+    public bool TryAddOption<T>(string name, T value)
```

Comment:
>public bool TryAddOption<T>(string name, T value) [](http://example.com/codeflow?start=4&length=49)

If this library is not compiled for ns2.0, you may use 
Dictionary<TKey, TValue>.TryAdd() instead?

Reply:
It is compiled for ns2.0 currently sadly.